### PR TITLE
Feature select package popup added to the scan order screen

### PR DESCRIPTION
**Pull request by Draakje10**
A select package popup that is able to select a package and amount of said package used

### CODE REVIEW

File: src/app/scan-order/models/selected-packaging.ts
Comment (DaaromThomas) [inline src/app/scan-order/models/selected-packaging.ts:1]: Packaging is not an array?

File: src/app/scan-order/scan-order.component.html
Comment (DaaromThomas) [inline src/app/scan-order/scan-order.component.html:1]: Maybe use @for? Ron said that is better

File: src/app/scan-order/scan-order.component.ts [hunk 1]
```diff
@@ -14,10 +16,10 @@ import {Product} from "./models/product";
 })
 export class ScanOrderComponent {
 
-  public orders: Order[] = [ new Order(13, "p", "asdf", 12)];
+  public orders: Order[] = [new Order(13, "p", "asdf", 12)];
   public products: Product[] = [
-      new Product("test", new Order(13, "p", "asdf", 12), "test product", 123, "test type"),
-      new Product( "test package", new Order(14, "pp", "test order", 13), "test product 2", 123456, "test product type")
+    new Product("test", new Order(13, "p", "asdf", 12), "test product", 123, "test type"),
```
Comment (DaaromThomas) [inline src/app/scan-order/scan-order.component.ts:21]: Use normal names for products and orders

File: src/app/scan-order/scan-order.component.html [hunk 2]
```diff
@@ -25,16 +26,17 @@
...
+        <th>packaging</th>
+        <th>amount</th>
+      </tr>
     </thead>
     <tbody>
-    <tr *ngFor="let package of packaging; index as i">
-      <td>{{package}}</td>
-      <td>2</td>
-    </tr>
+      <tr *ngFor="let package of packaging; index as i">
```
Comment (davidvdw02) [inline src/app/scan-order/scan-order.component.html:35]: i dont think you need index as i

File: src/app/scan-order/select-package-popup/select-package-popup.component.html [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    <div class="sidebar">
+        <img src="assets/package.png" alt="">
+    </div>
+    <div class="content-container">
+        <h2 mat-dialog-title>Change package</h2>
+        <mat-dialog-content>
+            <mat-form-field>
+                <mat-label>Select a package</mat-label>
+                <mat-select [(ngModel)]="selectedOption">
+                    <mat-option *ngFor="let package of packageList" [value]="package.id">
```
Comment (DaaromThomas) [inline src/app/scan-order/select-package-popup/select-package-popup.component.html:11]: Maybe use @for? Ron said it is better

File: src/app/scan-order/models/selected-packaging.ts
Comment (Draakje10) [reply src/app/scan-order/models/selected-packaging.ts:1]: Fixed with commit 68ec02f2284108f7276a4f55e68dda05dc4f6c0d

File: src/app/scan-order/scan-order.component.ts [hunk 4]
```diff
@@ -29,40 +31,49 @@ export class ScanOrderComponent {
...
+      this.errorMessage = "please scan a product";
+    } else if (isNaN(Number(this.InputOrderNumber))) {
       this.errorMessage = 'must be a valid ordernumber';
-    } else{
+    } else {
       this.errorMessage = '';
       this.scanOrderService.getProduct()
-          .subscribe((data: Product[]) => this.products = data);
+        .subscribe((data: Product[]) => this.products = data);
       console.log(this.orders);
```
Comment (davidvdw02) [inline src/app/scan-order/scan-order.component.ts:64]: console log?

File: src/app/scan-order/scan-order.component.ts [hunk 5]
```diff
@@ -29,40 +31,49 @@ export class ScanOrderComponent {
...
 
   }
 
+  openDialog() {
+    const dialogRef = this.dialog.open(SelectPackagePopupComponent, {
+      width: '750px',
+    });
+    dialogRef.afterClosed().subscribe(result => {
+      console.log('The dialog was closed', result.selectedPackaging.name, result.amount); // result heeft de gekozen package en de amount los als je er wat mee wilt doen kan dat vanaf hier
+    });
```
Comment (davidvdw02) [inline src/app/scan-order/scan-order.component.ts]: console.log? unnecessary comment

File: src/app/scan-order/select-package-popup/select-package-popup.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { SelectPackagePopupComponent } from './select-package-popup.component';
+
```
Comment (davidvdw02) [inline src/app/scan-order/select-package-popup/select-package-popup.component.spec.ts:4]: no tests?

File: src/app/scan-order/scan-order.component.ts (hunk 1 above)
Comment (Draakje10) [reply src/app/scan-order/scan-order.component.ts:21]: code is already in development

File: src/app/scan-order/scan-order.component.html (hunk 2 above)
Comment (Draakje10) [reply src/app/scan-order/scan-order.component.html]: Was already in development but fixed in commit 6c9a317b98101f065f859900d761aa805c0a7e60

**Pull request merged**